lib/parslet/atoms/can_flatten.rb

```diff
@@ -120,7 +120,7 @@ def flatten_repetition(list, named)
       return [] if named && list.empty?
 
       # If there are only strings, concatenate them and return that. 
-      foldl(list) { |s,e| s+e }
+      foldl(list) { |s,e| s.nil? ? e : e.nil? ? s : s + e }
```

Comment:
I realized that there's a better solution! Call `compact` on the list: `foldl(list.compact) { |s,e| s+e }`